src/Reflex/Dom/Widget/Input.hs

```diff
@@ -212,6 +214,110 @@ dropdown k0 options (DropdownConfig setK attrs) = do
+                          , _buttonGroupConfig_setValue     = never
+                          , _buttonGroupConfig_attributes   = constDyn mempty
+                          }
+
+buttonGroup :: (MonadWidget t m, Eq a, Show a)
+            => (Maybe Int -> Dynamic t a -> Dynamic t Bool -> m (Event t (), El t))
+            -> Dynamic t (Map.Map Int a)
+            -> ButtonGroupConfig t a
+            -> m (ButtonGroup t a)
+buttonGroup drawBtn dynBtns (ButtonGroupConfig pType pTag iVal setV dAtts) = do
+  dAtts' <- mapDyn (Map.insert "type" pType) dAtts
+  (parent, (dynV, internV, child)) <- elDynAttr' pTag dAtts' $ mdo
+    pb <- getPostBuild
+    let externSet = attachWith revLookup (current dynBtns) setV
+        initSet   = attachWith revLookup (current dynBtns) (iVal <$ pb)
+        internSet = leftmost [initSet, clickSelEvts]
+        internVal = attachWith (\m k -> k >>= flip Map.lookup m)
+                              (current dynBtns)
+                              internSet
+    dynK     <- holdDyn Nothing $ leftmost [internSet, externSet]
+    dynBtns' <- mapDyn (Map.mapKeys Just) dynBtns
+    (clickSelEvts, children) <- selectViewListWithKey_' dynK dynBtns' drawBtn
+    dynSelV <- combineDyn (\k m -> k >>= flip Map.lookup m) dynK dynBtns
+    return (dynSelV, internVal, children)
+  return (ButtonGroup { _buttonGroup_value    = dynV
+                      , _buttonGroup_select   = internV
+                      , _buttonGroup_element  = parent
+                      , _buttonGroup_children = child
+                      })
+ 
+revLookup :: Eq a => Map.Map Int a -> Maybe a -> Maybe Int
+revLookup _ Nothing = Nothing
+revLookup m (Just v) = listToMaybe . Map.keys $ Map.filter (== v) m
+
+selectViewListWithKey_' :: forall t m a v.(MonadWidget t m)
+                        => Dynamic t (Maybe Int)
+                        -> Dynamic t (Map.Map (Maybe Int) v)
+                        -> (Maybe Int -> Dynamic t v -> Dynamic t Bool -> m (Event t a, El t))
+                        -> m (Event t (Maybe Int), Dynamic t (Map.Map Int (El t)))
+selectViewListWithKey_' selection vals mkChild = do
+  let selectionDemux = demux selection
+  selectChildAndEl <- listWithKey vals $ \k v -> do
+    selected <- getDemuxed selectionDemux k
+    (selectSelf, selfEl) <- mkChild k v selected
+    return $ (fmap (const k) selectSelf, selfEl)
+  selectChild <- mapDyn (Map.map fst) selectChildAndEl
+  els <- mapDyn (Map.mapKeys fromJust 
+                . Map.mapMaybeWithKey (\k v -> maybe Nothing (const (Just v)) k)
+                . Map.map snd) selectChildAndEl
+  selEvents <- liftM switchPromptlyDyn $ mapDyn (leftmost . Map.elems) selectChild
+  return (selEvents, els)
+
+
+radioButtons :: (MonadWidget t m, Eq a, Show a) 
+             => Dynamic t String
+             -> Dynamic t [(a, String)]
+             -> ButtonGroupConfig t a
+             -> m (ButtonGroup t a)
+radioButtons dynName dynElems bgConfig0 = do
```

Comment:
It's too bad we have to ask the user to pick a name here, but I can't see a way around it without having some infrastructure for allocating fresh names.  Let's add a big warning here, though, since the widget will break (i think) if the user picks a non-unique name.